### PR TITLE
✨ NEW: Support tuple access in guards

### LINKED ISSUE #724: Support tuple access in guards
```rust
case x {
  [first, ..rest] if first.0 == key -> "ok"
  _ -> "ko"
}
```

### PR DESCRIPTION
**Pull request by mrgleam**
Close #724 

Hi @lpil 

Thank you for your help. Please review my code. I make it work for tuple access in guards but I has one problem about lalrpop.

I found error when I try to remove annotation #[inline]
Screen Shot 2563-10-07 at 17 37 48

### CODE REVIEW

File: src/grammar.lalrpop [hunk 1]
```diff
@@ -499,9 +499,17 @@ ClauseGuard4: UntypedClauseGuard = {
...
+        location: location(s, e),
+        index,
+        typ: (),
+        tuple: Box::new(name),
+    },
+
     ClauseGuard5 => <>,
 }
 
+#[inline]
```
Comment (lpil) [inline src/grammar.lalrpop:512]: This is the first time I'm learning about `inline`. Do you know where I can find the documentation for it? I always struggle with finding info about lalrpop.

**Review by lpil — COMMENTED** [review]
This all looks really good and very clean! Thank you!

That lalrpop issue is a bit confusing to me, honestly I'm not the best with lalrpop. I'm going to take a little time and try to get an understanding of what is happening here.

If I can't figure it out maybe we can merge this anyway. After all, all the tests are passing. :)

File: src/grammar.lalrpop (hunk 1 above)
Comment (mrgleam) [reply src/grammar.lalrpop:512]: i found this blog http://smallcultfollowing.com/babysteps/blog/2016/03/02/nice-errors-in-lalrpop/

Comment (lpil) [reply src/grammar.lalrpop:512]: Ah nice! Thank you

**Review by lpil — APPROVED** [review]
OK cool I'm happy. 👍 I may fiddle with the grammar a bit after merge just so I can learn more about this behaviour.

Could you update the changelog file please?

**Comment by mrgleam** [comment]
> OK cool I'm happy. 👍 I may fiddle with the grammar a bit after merge just so I can learn more about this behaviour.
> 
> Could you update the changelog file please?

yes. updated 👍

**Review by lpil — APPROVED** [review]
Thank you!

**Pull request merged**